pkg/migration/azure_crs.go

```diff
@@ -391,6 +396,158 @@ func (m *azureMigrator) readRelease(ctx context.Context, ver string) error {
 	return nil
 }
 
+func (m *azureMigrator) updateCluster(ctx context.Context) error {
+	cluster := m.crs.cluster
+
+	// Drop operator version label.
+	delete(cluster.Labels, label.AzureOperatorVersion)
+
+	// Drop finalizers.
+	cluster.Finalizers = nil
+
+	// Adjust k8s apiserver bind port to match kubeadm.
+	if cluster.Spec.ClusterNetwork != nil && cluster.Spec.ClusterNetwork.APIServerPort != nil {
+		cluster.Spec.ClusterNetwork.APIServerPort = to.Int32Ptr(6443)
+	}
+
+	cluster.Spec.ControlPlaneRef = &corev1.ObjectReference{
+		APIVersion: m.crs.kubeadmControlPlane.APIVersion,
+		Kind:       m.crs.kubeadmControlPlane.Kind,
+		Name:       m.crs.kubeadmControlPlane.Name,
+	}
+
+	err := m.mcCtrlClient.Update(ctx, cluster)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	return nil
+}
+
+func (m *azureMigrator) updateAzureCluster(ctx context.Context) error {
+	cluster := m.crs.azureCluster
+
+	// Drop operator version label.
+	delete(cluster.Labels, label.AzureOperatorVersion)
+
+	// Drop finalizers.
+	cluster.Finalizers = nil
+
+	// Use default credentials.
+	cluster.Spec.IdentityRef = nil
+
+	if cluster.Spec.NetworkSpec.APIServerLB.Name == "" {
+		cluster.Spec.NetworkSpec.APIServerLB = capz.LoadBalancerSpec{
+			Name: fmt.Sprintf("%s-%s-%s", cluster.Name, "API", "PublicLoadBalancer"),
+			SKU:  "Standard",
+			Type: "Public",
+			FrontendIPs: []capz.FrontendIP{
+				{
+					Name: fmt.Sprintf("%s-%s-%s-%s", cluster.Name, "API", "PublicLoadBalancer", "Frontend"),
+					PublicIP: &capz.PublicIPSpec{
+						Name: fmt.Sprintf("%s-%s-%s-%s", cluster.Name, "API", "PublicLoadBalancer", "PublicIP"),
+					},
+				},
+			},
+		}
+	}
+
+	// START OF UGLY HACK **************************************************
+	// XXX: This is just a shortcut to make testing ergonomics better. Final
+	// code should be rewritten.
+	if len(cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs) == 0 {
+		ip := capz.FrontendIP{
+			Name: fmt.Sprintf("%s-%s-%s-%s", cluster.Name, "API", "PublicLoadBalancer", "Frontend"),
+			PublicIP: &capz.PublicIPSpec{
+				Name: fmt.Sprintf("%s-%s-%s-%s", cluster.Name, "API", "PublicLoadBalancer", "PublicIP"),
+			},
+		}
+
+		cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs = append(cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs, ip)
```

Comment:
Following ifs will cover the rest.

```suggestion
		cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs = append(cluster.Spec.NetworkSpec.APIServerLB.FrontendIPs, capz.FrontendIP{})
```